bin/clang/astVisitor.cc

```diff
@@ -212,12 +212,23 @@ CompilerGlobals::setup(clang::CompilerInstance* CI)
 
 
   if (includeListOpt.getNumOccurrences()){
+    char fullpathBuffer[1024];
     StringRef sref(includeListOpt);
     std::istringstream sstr(sref.str());
     std::string path;
     while (std::getline(sstr, path, ':'))
     {
       includePaths.push_back(path);
+      //make sure the full paths (following symlinks)
+      //are included for checking
+      const char* fullpath_cstr = realpath(path.c_str(), fullpathBuffer);
+      if (fullpath_cstr){
+        std::string fullpath = fullpath_cstr;
+        //if we follow symlinks or remove rel path and is different
+        if (fullpath != path){
+          includePaths.push_back(fullpath);
+        }
+      }
```

Comment:
The way that I read realpath as working, means that we should just drop line 221 and just always push_back fullpath?? And error if fullpath_cstr is null? 

Even better we can emplace_back fullpath_cstr and not require the explicit conversion. Hopefully I didn't check this.

Reply:
Just for safety's sake, if there is any checking of paths done. I want to make sure we catch either the symlink path or the fundamental path.